src/libopensc/pkcs15-cert.c

```diff
@@ -273,26 +273,27 @@ sc_pkcs15_get_extension(struct sc_context *ctx, struct sc_pkcs15_cert *cert,
 			LOG_FUNC_RETURN(ctx, r);
 
 		/* is it the RN we are looking for */
-		if(sc_compare_oid(&oid, type) != 0) {
+		if (sc_compare_oid(&oid, type) != 0) {
 			if (*ext_val == NULL) {
 				*ext_val= val;
 				val = NULL;
 				*ext_val_len = val_len;
+				/* do not free here -- leaking the allocated value */
```

Comment:
I see. Maybe rephrasing this to something like `do not free here -- return the allocated value to caller` would be more appropriate.